src/main/java/io/cdap/e2e/pages/locators/CdfNameSpaceAdminLocators.java

```diff
@@ -0,0 +1,160 @@
+  }
+
+  public static WebElement locateValueProperty(String valueProperty, int row) {
+    String xpath = "//*[@data-cy='" + valueProperty + "" + row + "']//input[@placeholder='value']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  public static WebElement locateAddRowButtonProperty(String addRowProperty, int row) {
+    String xpath =
+        "//*[@data-cy='" + addRowProperty + "" + row + "']//span"
+            + "//button[@type='submit' and @class='btn add-row-btn btn-link']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  @FindBy(how = How.XPATH, using = "//*[@class='namespace-dropdown']")
+  public static WebElement namespaceDropdown;
+
+  @FindBy(how = How.XPATH, using = "//div[contains(text(),'Add Namespace')]")
+  public static WebElement addNamespace;
+
+  @FindBy(how = How.XPATH, using = "//button[@data-cy='wizard-next-btn']")
+  public static WebElement clickOnNextButton;
+
+  @FindBy(how = How.XPATH, using = "//button[@data-cy='wizard-finish-btn']")
+  public static WebElement clickOnFinishButton;
+
+  public static WebElement locateKeyProperty(int row) {
+    String xpath = "//*[@data-cy= 'key-value-pair-" + row + "']//input[@placeholder='key']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  public static WebElement locateValueProperty(int row) {
+    String xpath = "//*[@data-cy= 'key-value-pair-" + row + "']//input[@placeholder='value']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  public static WebElement locateAddRowButtonProperty(int row) {
+    String xpath = "//*[@data-cy= 'key-value-pair-" + row + "']//span//button[@type='submit' and " +
+        "@class='btn add-row-btn btn-link']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  public static WebElement locateProvisionerInList(String provisionerName) {
+    return SeleniumDriver.getDriver()
+        .findElement(By.xpath("//div[@data-cy='" + provisionerName + "']"));
+  }
+
+  public static WebElement locateButtonType(String buttonType) {
+    return SeleniumDriver.getDriver()
+        .findElement(By.xpath("//button[@data-cy='" + buttonType + "']"));
+  }
+
+  public static By profilePropertiesPage() {
+    return By.xpath("//h5[contains(text(), 'Create a profile')]");
+  }
+
+  public static By locateProfileTitle(String profileName) {
+    return By.xpath("//div[@title='" + profileName + "']");
+  }
+
+  @FindBy(how = How.XPATH, using = "//a[contains(text(),\"Switch to\")]")
+  public static WebElement switchToCreatedNamespace;
+
+  @FindBy(how = How.XPATH, using = "//a[contains(text(),'Go to homepage')]")
+  public static WebElement goToHomepage;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy='wizard-result-icon-close-btn']")
+  public static WebElement closeAddNameSpaceWindow;
+
+  @FindBy(how = How.XPATH, using = "//span[@id='setpreferences-fd']")
```

Comment:
prefer `data-testid` over `id`, as it indicates that the testid is used in e2e tests and should not be changed.